src/random_unitary.jl

```diff
@@ -0,0 +1,52 @@
+using MatrixAlgebraKit: qr_full!
+using Random: Random, AbstractRNG, randn!
+
+function square_zero_map(elt::Type, ax::Tuple{AbstractUnitRange,Vararg{AbstractUnitRange}})
+  return zeros(elt, (ax..., ax...))
+end
+
+function random_unitary!(rng::AbstractRNG, a::AbstractArray)
+  @assert iseven(ndims(a))
+  ndims_codomain = ndims(a) ÷ 2
+  biperm = blockedperm(ntuple(identity, ndims(a)), (ndims_codomain, ndims_codomain))
+  a_mat = fusedims(a, biperm)
+  random_unitary!(rng, a_mat)
+  splitdims!(a, a_mat, biperm)
+  return a
+end
+
+function random_unitary!(rng::AbstractRNG, a::AbstractMatrix)
+  Q, _ = qr_full!(randn!(rng, a); positive=true)
+  a .= Q
+  return a
+end
+
+function random_unitary(
+  rng::AbstractRNG, elt::Type, ax::Tuple{AbstractUnitRange,Vararg{AbstractUnitRange}}
+)
+  return random_unitary!(rng, square_zero_map(elt, ax))
+end
+
+# Canonicalizing other kinds of inputs.
+function random_unitary(
+  rng::AbstractRNG, elt::Type, dims::Tuple{Vararg{Union{AbstractUnitRange,Integer}}}
+)
+  return random_unitary(Random.default_rng(), elt, map(to_axis, dims))
```

Comment:
Is it a mistake that the `rng` argument isn't used here? Should that argument be dropped and/or used inside the function?

Reply:
Yes, that's a mistake, thanks for catching that.